sqlglot/dialects/clickhouse.py

```diff
@@ -8,6 +8,8 @@
     inline_array_sql,
     no_pivot_sql,
     rename_func,
+    safeconcat_sql,
```

Comment:
this always goes hand in hand, should we do a dialect flag instead?

Reply:
refactor strict cast as well if it’s rasy

Reply:
Refactored this into a dialect flag called `strict_string_concat`